kafka-client/src/main/java/dev/responsive/utils/SubPartitioner.java

```diff
@@ -47,14 +50,39 @@ public class SubPartitioner {
   // number of sub partitions is equal to the number of original
   // partitions, all keys would be mapped to the same sub-partition
   // and all other sub-partitions would be empty
-  private static final int SALT = 31;
+  static final int SALT = 31;
 
   /**
    * the number of subpartitions is: {@code original_partitions * n}
    */
   private final int factor;
   private final Function<Bytes, Integer> hasher;
 
+  public static SubPartitioner create(
+      final OptionalInt actualRemoteCount,
+      final int kafkaPartitions,
+      final int desiredNum,
+      final TableName name,
+      final String changelogTopicName
+  ) {
+    final int factor = (desiredNum == ResponsiveConfig.NO_SUBPARTITIONS) ? 1 : desiredNum / kafkaPartitions;
```

Comment:
Should we err on the side of over-partitioning instead? Also, consider including this formula in the doc.